src/gui/attributetable/qgsdualview.cpp

```diff
@@ -133,10 +136,26 @@ void QgsDualView::init( QgsVectorLayer *layer, QgsMapCanvas *mapCanvas, const Qg
 
   mTableView->setModel( mFilterModel );
   mFeatureListView->setModel( mFeatureListModel );
-  delete mAttributeForm;
-  mAttributeForm = new QgsAttributeForm( mLayer, mTempAttributeFormFeature, mEditorContext );
-  mTempAttributeFormFeature = QgsFeature();
-  if ( !context.parentContext() )
+
+  connect( mFilterModel, &QgsAttributeTableFilterModel::sortColumnChanged, this, &QgsDualView::onSortColumnChanged );
+
+  if ( mFeatureListPreviewButton->defaultAction() )
+    mFeatureListView->setDisplayExpression( mDisplayExpression );
+  else
+    columnBoxInit();
+
+  // This slows down load of the attribute table heaps and uses loads of memory.
+  //mTableView->resizeColumnsToContents();
+
+  if ( mFeatureListModel->rowCount( ) > 0 )
+    mFeatureListView->setEditSelection( QgsFeatureIds() << mFeatureListModel->data( mFeatureListModel->index( 0, 0 ), QgsFeatureListModel::Role::FeatureRole ).value<QgsFeature>().id() );
+
+}
+
+void QgsDualView::initAttributeForm( const QgsFeature &feature )
+{
+  mAttributeForm = new QgsAttributeForm( mLayer, feature, mEditorContext );
```

Comment:
Can you insert a `Q_ASSERT( !mAttributeForm )` one line before?

Reply:
Yeah, in fact, I've moded this function to be private, it shouldn't be called from outside. Added the Q_ASSERT too.